### ISSUE TITLE
Allow user to set a location as default

### ISSUE THREAD

**Issue by dooliddler** [WhosAble/WhosAble#68]
When creating a job and setting a location for services to be preformed, allow a user to set a default location such as a home for babysitting or an address for a business location.  If a location is set as default, new jobs will be pre populated with this address unless the user changes it.

**Comment by mahcloud** [comment]
How would you like this to look in the interface?

**Comment by dooliddler** [comment]
What are my options?   Push it to heroku?

**Comment by dooliddler** [comment]
Disregard.  I misread.

**Comment by dooliddler** [comment]
A "set this as default location" checkbox

**Comment by mahcloud** [comment]
would this checkbox be on the location form page?

**Comment by mahcloud** [comment]
What would happen if the user unchecks the default location checkbox?
My concern is if there is no default how would that effect the job form. Or if that is possible.

**Comment by dooliddler** [comment]
Good question.  I am a new user and I create a job (such as babysitting, location home address).  when I return to the app to create another babysitting job and specify the same location, I will click a button that will take me to a page to select the location.  In the case above, I would have only one previous location to select.  If I wanted to select this as my default address, when creating a new job, this address would be there by default.  If I were to uncheck the default address box, when I return to the app to create another babysitting job, I will click a button that will take me to a page to select the location.

Clear?  Makes sense?

**Comment by dooliddler** [comment]
When you create a new address, it would offer to make this the default.

**Comment by mahcloud** [comment]
So to reiterate, If a default is not or set or unset, then the user is presented with no selected location when they first hit the new job form?

**Comment by dooliddler** [comment]
Isn't that how it functions now?

**Comment by mahcloud** [comment]
No, currently it just takes the first from the list of locations.
If we add a checkbox to the edit location form, the user could uncheck the defaulted location. The user would then end up with no default location. Do we want to prevent this?

**Comment by dooliddler** [comment]
Essentially, drawing from the top of the list accomplishes the same thing.  I'm just trying to save a step if multiple locations are added.  

If we add a 'set as default location' feature and they uncheck leaving none selected as default, it could then drag from top of list.

Do you think this is necessary?

**Comment by mahcloud** [comment]
it's an edge case for sure, but as a programmer we have to program it one way or the other. I just want to make sure I'm doing it the way you'd prefer the first time.

**Comment by dooliddler** [comment]
Should each added address then have a 'set as default location' check box?  Only one can be set as default so selecting one overrides an existing default.

**Comment by mahcloud** [comment]
Well I think there are quite a few ways to skin this cat.

- The check box idea you have been suggesting.
- drop down list of addresses that can be selected.
- Change default location link on page adds a link on each location that can be clicked to change the default.

If we go with the check box, then yes checking the box and saving would override the current default.

**Comment by dooliddler** [comment]
I don't understand the difference between 1 & 3 exactly.  #2 would work also.
